tests/e2e/unified_test.go

```diff
@@ -622,8 +622,100 @@ func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture reso
+								// Include settings.ipConfiguration.authorizedNetworks in response, even if it's empty.
+								var val []string
+								if err := unstructured.SetNestedStringSlice(obj, val, "settings", "ipConfiguration", "authorizedNetworks"); err != nil {
+									t.Fatal(err)
+								}
+							}
+						}
+						if _, found, _ := unstructured.NestedString(obj, "gceZone"); found {
+							// Hardcode the zone. GCP chooses this zone within the
+							// region, and it varies based on availability.
+							if err := unstructured.SetNestedField(obj, "us-central1-a", "gceZone"); err != nil {
+								t.Fatal(err)
+							}
+						}
+						if ipConfig, found, _ := unstructured.NestedMap(obj, "settings", "ipConfiguration"); found {
+							// Hack fix: remove unpublished field that's suddenly showing up in real gcp proto responses.
+							delete(ipConfig, "serverCaMode")
+							if err := unstructured.SetNestedMap(obj, ipConfig, "settings", "ipConfiguration"); err != nil {
+								t.Fatal(err)
+							}
+						}
+					})
+					jsonMutators = append(jsonMutators, func(obj map[string]any) {
+						if val, found, err := unstructured.NestedString(obj, "kind"); err != nil || !found || val != "sql#usersList" {
+							// Only run this mutator for sql users list objects.
+							return
+						}
+						if items, found, _ := unstructured.NestedSlice(obj, "items"); found {
+							// Include items[].host in response, even if it's empty.
+							newItems := []interface{}{}
+							for _, item := range items {
+								if itemMap, ok := item.(map[string]interface{}); ok {
+									if _, found, _ := unstructured.NestedStringSlice(itemMap, "host"); !found {
+										if err := unstructured.SetNestedField(itemMap, "", "host"); err != nil {
+											t.Fatal(err)
+										}
+									}
+									newItems = append(newItems, itemMap)
+								}
+							}
+							if err := unstructured.SetNestedSlice(obj, newItems, "items"); err != nil {
+								t.Fatal(err)
+							}
+						}
+					})
+					jsonMutators = append(jsonMutators, func(obj map[string]any) {
+						if val, found, err := unstructured.NestedString(obj, "kind"); err != nil || !found || val != "sql#operation" {
+							// Only run this mutator for sql operation objects.
+							return
+						}
+						if _, found, _ := unstructured.NestedString(obj, "name"); found {
+							// Set a hardcoded UUID.
+							if err := unstructured.SetNestedField(obj, "11111111-aaaa-2222-bbbb-123456789abc", "name"); err != nil {
+								t.Fatal(err)
+							}
+						}
+						if _, found, _ := unstructured.NestedString(obj, "selfLink"); found {
+							// Set a hardcoded self link (based on name).
+							if err := unstructured.SetNestedField(obj, "https://sqladmin.googleapis.com/sql/v1beta4/projects/${projectId}/operations/11111111-aaaa-2222-bbbb-123456789abc", "selfLink"); err != nil {
+								t.Fatal(err)
```

Comment:
I think we would want a more generic normalize function for self link: it can replace the ID to something like ${uniqueId} for all resources that has a selflink in the response.  Check `extractIDsFromLinks`

Reply:
Ah, good point. I think we handle this already. It may have been pointless to add this normalizer. Let me double-check.

Reply:
Ok, fixed. Added another check for sql operation ids. Thanks :)